aws-amplify/docs
# Update hompage

#### Description of changes:
- Updated `FeatureItem` to use `next/link` so home page links could use nextjs client transitions and preloading
- Updated the home page to have the gen2 content
- Updated the platform index pages to have the correct links

Staging site: https://gen2-homepage.d1ywzrxfkb9wgg.amplifyapp.com/

To test:
1. Go to https://gen2-homepage.d1ywzrxfkb9wgg.amplifyapp.com/
2. Verify the bottom half of the page shows the gen2 content from https://docs.amplify.aws/gen2/
3. Go to https://gen2-homepage.d1ywzrxfkb9wgg.amplifyapp.com/javascript
4. Verify the bottom half of the page shows the gen2 content from https://docs.amplify.aws/gen2/

#### Related GitHub issue #, if available:

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?

Which platform(s) are affected by this PR (if applicable)?

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks




      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._
